src/utils/floor.ts

```diff
@@ -0,0 +1,538 @@
+
+  // Initialize IRoomNode variable (adjRooms.type is automatically assigned during initialization with the values of 'floor', which is a
+  // 'ROOM_TYPE' type)
+  const adjRooms: IRoomNode[][] = floor.map((row) =>
+    row.map((roomType) => {
+      const adjRoom: IRoomNode = {
+        type: roomType,
+        explored: false,
+        eastDoor: false,
+        westDoor: false,
+        northDoor: false,
+        southDoor: false,
+      };
+      return adjRoom;
+    })
+  );
+
+  // Check for each room, whether it has adjacent rooms in each cardinal direction.
+  for (let row = 0; row < floor.length; row++) {
+    for (let col = 0; col < floor.length; col++) {
+      // Check Right/East
+      if (
+        adjRooms[row + dir[0][0]][col + dir[0][1]].type !== ROOM_TYPE.NULL || // The room isn't ROOM_TYPE NULL.
+        (row + dir[0][0] >= 0 && // The room to the right/east is within the bounds of the grid
+          col + dir[0][1] >= 0 &&
+          row + dir[0][0] < floor.length &&
+          col + dir[0][1] < floor.length)
+      ) {
+        adjRooms[row + dir[0][0]][col + dir[0][1]].eastDoor = true;
+      }
+
+      // Check Left/West
+      if (
+        adjRooms[row + dir[1][0]][col + dir[1][1]].type !== ROOM_TYPE.NULL || // The room isn't ROOM_TYPE NULL.
+        (row + dir[1][0] >= 0 && // The room to the left/west is within the bounds of the grid
+          col + dir[1][1] >= 0 &&
+          row + dir[1][0] < floor.length &&
+          col + dir[1][1] < floor.length)
+      ) {
+        adjRooms[row + dir[1][0]][col + dir[1][1]].westDoor = true;
+      }
+
+      // Check Down/South
+      if (
+        adjRooms[row + dir[2][0]][col + dir[2][1]].type !== ROOM_TYPE.NULL || // The room isn't ROOM_TYPE NULL.
+        (row + dir[2][0] >= 0 && // The room down/south is within the bounds of the grid
+          col + dir[2][1] >= 0 &&
+          row + dir[2][0] < floor.length &&
+          col + dir[2][1] < floor.length)
+      ) {
+        adjRooms[row + dir[2][0]][col + dir[2][1]].southDoor = true;
+      }
+
+      // Check Up/North
+      if (
+        adjRooms[row + dir[3][0]][col + dir[3][1]].type !== ROOM_TYPE.NULL || // The room isn't ROOM_TYPE NULL.
+        (row + dir[3][0] >= 0 && // The room up/north is within the bounds of the grid
+          col + dir[3][1] >= 0 &&
+          row + dir[3][0] < floor.length &&
+          col + dir[3][1] < floor.length)
+      ) {
+        adjRooms[row + dir[3][0]][col + dir[3][1]].northDoor = true;
+      }
+
+      adjRooms[row][col].explored = true; // Room has been explored.
+    }
+  }
+
+  // const adjRooms: IRoomNode[][] = Array.from({ length: floor.length }, () =>
+  //   Array(floor[0].length).fill({
+  //     type: floor[0][0],
+  //     explored: false,
+  //     eastDoor: false,
+  //     westDoor: false,
+  //     northDoor: false,
+  //     southDoor: false,
+  //   })
+  // );
+  // room.explored = true
+
+  return adjRooms;
+}
```

Comment:
Awesome job. All that's left is to make some unit tests for generateFloorPlan and findAdjacentRooms. I would also recommend renaming findAdjacentRooms to connectAdjacentRooms instead.